packages/@aws-cdk/integ-tests/lib/assertions/providers/lambda-handler/assertion.ts

```diff
@@ -1,34 +1,63 @@
 /* eslint-disable no-console */
-import * as assert from 'assert';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { Match } from '@aws-cdk/assertions/lib/helpers-internal';
 import { CustomResourceHandler } from './base';
-import { AssertionRequest, AssertionResult } from './types';
+import { AssertionResult, AssertionRequest } from './types';
 
 export class AssertionHandler extends CustomResourceHandler<AssertionRequest, AssertionResult> {
   protected async processEvent(request: AssertionRequest): Promise<AssertionResult | undefined> {
+    let actual = request.actual;
+    const expected = decodeCall(request.expected);
     let result: AssertionResult;
+    let matcher;
+    console.log(`Testing equality between ${JSON.stringify(request.actual)} and ${JSON.stringify(request.expected)}`);
+
     switch (request.assertionType) {
```

Comment:
Instead of having a single `assertionType`, could we do something arbitrarily recursive? We could encode it the same way as CloudFormation intrinsics: Something like:

```js
{ '$ObjectLike': {
  Message: [
    Body: { 
      'Match::Like': {
        Elements: { '$ArrayWith': [{ Asdf: 3 }] },
      }
    }
  ]
}}
```

Then in the handler we'd recursively turn that structure into matchers and arguments to matchers before executing.

On the construction side, we'd have helpers to construct this JSON so consumers don't have to see the ugliness:

```ts
export class Match {
  public static arrayWith(xs: string[]): any {
    return { '$ArrayWith': xs };
  }
}
```

Reply:
Added. Let me know what you think about the API.